src/Lifecycle.js

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2015, 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import MatrixClientPeg from './MatrixClientPeg';
+import Notifier from './Notifier'
+import UserActivity from './UserActivity';
+import Presence from './Presence';
+import dis from './dispatcher';
+
+function login(credentials, options) {
+    credentials.guest = Boolean(credentials.guest);
+    console.log("onLoggedIn => %s (guest=%s)", credentials.userId, credentials.guest);
+    MatrixClientPeg.replaceUsingAccessToken(
+        credentials.homeserverUrl, credentials.identityServerUrl,
+        credentials.userId, credentials.accessToken, credentials.guest
+    );
+
+    dis.dispatch({action: 'on_logged_in'});
+
+    startMatrixClient(options);
+}
+
+function logout() {
+    if (MatrixClientPeg.get().isGuest()) {
+        // logout doesn't work for guest sessions
+        // Also we sometimes want to re-log in a guest session
+        // if we abort the login
+        _onLoggedOut();
+        return;
+    }
+
+    return MatrixClientPeg.get().logout().then(_onLoggedOut,
+        // Just throwing an error here is going to be very unhelpful
```

Comment:
Is this a thing we're doing, btw? console logging in webapps is generally considered poor form in production.